### DISCUSSION TITLE
Is CEL supports loops or iterations to run certain logic repeatedly ??   similar to "while loops" in general programing languages ??

### DISCUSSION THREAD

**Discussion by krishna-macharla** [google/cel-spec#421] [General]
Is CEL supports loops or iterations to run certain logic repeatedly ??   similar to "while loops" in general programing languages ??
We want to implement logic for pagination for API calls.  For that we need to run some logic in loops .
Please suggest how can we write this logic in CEL.

**Comment by CLOVIS-AI** [comment]
CEL isn't a programming language, and it doesn't _do_ things. SQL doesn't have loops, CEL doesn't either, for similar reasons.

Instead, CEL can transform data, for example with `map` or `filter`: https://github.com/google/cel-spec/blob/master/doc/langdef.md#macros

But if your goal is pagination, you probably just want to add an additional parameter to your CEL query like `page = `, and do paging on the backend with your regular programming language.

**Reply by krishna-macharla** [reply]
Thank you for quick response. unfortunately, backend is third-party and they not supporting pagination.
